### ISSUE TITLE
Fix linting issues caused due to methods with too many return statements

### ISSUE THREAD

**Issue by angrezichatterbox** [scribe-org/Scribe-Android#92]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Methods with too many return statements can be hard to read. Aim to have a single point of exit from a method, which makes the code more understandable and easier to debug.

Before fixing:

```
fun getDiscount(category: String): Int {
    if (category == "A") return 10
    if (category == "B") return 20
    if (category == "C") return 30
    return 0
}

```

After fixing:

```
fun getDiscount(category: String): Int {
    return when (category) {
        "A" -> 10
        "B" -> 20
        "C" -> 30
        else -> 0
    }
}

```

More about this can be read from the [docs](https://detekt.dev/docs/rules/style/#returncount)

These can be identified across the codebase by the following steps:

- Enable the ReturnCount rule in detekt.yml.
- Run ./gradlew detekt to identify methods with too many return statements.
- Refactor methods to reduce the number of return statements.

### Contribution

I would love to work on this and am more than willing to help anyone solve this issue. 😄

**Comment by KesharwaniArpita** [comment]
Can I be assigned to fix this issue?

**Comment by angrezichatterbox** [comment]
> Can I be assigned to fix this issue?

Assigned :)  Thanks for taking up the issue.

**Comment by andrewtavis** [comment]
Closed by #191 🚀 Really great to have the continued progress on linting! Thanks @angrezichatterbox and @KesharwaniArpita 😊
